### PR TITLE
[4.0] Remove edit button from toolbar

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

As it stands, there's no reason to have an edit button in the toolbar. Clicking on the article title directs you to the edit view.

Survey results provided y the UX team a while back shows that the vast majority of users click on the article title to edit the content as opposed to using the toolbar.

This PR removes the edit button and add a small icon to the left of the title, like so:

toolbar

I've only done this for com_content (Article Manager) as a start, so will do the rest once approved.

### Progress
- [x] com_content
- [x] rest of components

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on f481f3ee2de1de873aa8df3e2743a630746913df<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17091">issues.joomla.org/tracker/joomla-cms/17091</a>.</sub>

**Comment by brianteeman** [comment]
Looks great#

off the top of my head you need to check to make sure everything is ok when an item is checked out and also if there is an item that you dont have permission to edit.

File: administrator/components/com_content/tmpl/articles/default.php [hunk 1]
```diff
@@ -167,7 +167,7 @@
 										<?php endif; ?>
 										<?php if ($canEdit || $canEditOwn) : ?>
 											<a class="hasTooltip" href="<?php echo JRoute::_('index.php?option=com_content&task=article.edit&id=' . $item->id); ?>" title="<?php echo JText::_('JACTION_EDIT'); ?>">
-												<?php echo $this->escape($item->title); ?></a>
+												<span class="fa fa-pencil-square mr-2"></span><?php echo $this->escape($item->title); ?></a>
```
Comment (brianteeman) [inline administrator/components/com_content/tmpl/articles/default.php]: aria hidden=true 
please

File: administrator/components/com_content/tmpl/featured/default.php [hunk 2]
```diff
@@ -158,7 +158,7 @@
 										<?php endif; ?>
 										<?php if ($canEdit) : ?>
 											<a class="hasTooltip" href="<?php echo JRoute::_('index.php?option=com_content&task=article.edit&return=featured&id=' . $item->id); ?>" title="<?php echo JText::_('JACTION_EDIT'); ?>">
-												<?php echo $this->escape($item->title); ?></a>
+												<span class="fa fa-pencil-square mr-2"></span><?php echo $this->escape($item->title); ?></a>
```
Comment (brianteeman) [inline administrator/components/com_content/tmpl/featured/default.php]: aria hidden = true please

**Comment by C-Lodder** [comment]
`aria-hidden="true"` added to icon.

The icon now also only shows when the item is checked in

toolbar2

**Comment by brianteeman** [comment]
Thanks - seems great - will test in depth later today

**Comment by brianteeman** [comment]
I am wondering if we can improve this for accessibility

Currently it is two links and says "edit" "edit"

It would be much better if it was just one link that says "edit title"

**Comment by brianteeman** [comment]
I am assuming you will do another PR to remove the edit button from all the other components?

**Comment by C-Lodder** [comment]
Technically it's still one link as both the icon and title with within the same `<a>` tag. I just added a right margin to the icon so that when you ho er over the title, the underline doesnt appear yhrougj the gap.

I'll do it all as 1 PR but want to make sure people are happy with the initial concept first

**Comment by brianteeman** [comment]
Quite right i should have looked atthe code not just the output.

What you could do though is to change the line to something like

`<a class="hasTooltip" href="<?php echo JRoute::_('index.php?option=com_content&task=article.edit&id=' . $item->id); ?>" title="<?php echo JText::_('JACTION_EDIT'); ?> <?php echo $this->escape(addslashes($item->title)); ?>">
`

And then the title will be more meaningful as "Edit Article Name"

**Comment by C-Lodder** [comment]
@brianteeman updated

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 76fed33692caa4ac5a0ce1517b18d865cc3d8b6b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17091">issues.joomla.org/tracker/joomla-cms/17091</a>.</sub>

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 76fed33692caa4ac5a0ce1517b18d865cc3d8b6b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17091">issues.joomla.org/tracker/joomla-cms/17091</a>.</sub>

**Comment by dgrammatiko** [comment]
> What you could do though is to change the line to something like

Not with tooltips. Links are one of the fundamental concepts of the web, if ppl don't know that clicking on a link will take them to another page (in this case the edit view) we can't and shouldn't try to teach them in the backend about it!

**Comment by brianteeman** [comment]
It's an a11y thing where you are explicitly saying what they will edit. Otherwise you have a hundred links on the page with the same title

**Comment by dgrammatiko** [comment]
Then let's prepend the title with something like `Edit: Title of the article`. No tooltips, no aria-whatever

**Comment by mbabker** [comment]
Isn't that what the icon is communicating to visually capable users?  Or is it better to have a list of 20 links that reads `Edit: Foo`?

**Comment by brianteeman** [comment]
@dgt41 that's exactly what the code I proposed does. Doesn't it?

**Comment by dgrammatiko** [comment]
@mbabker @brianteeman the icon itself is not ok, we need a fallback for screenreaders/ppl visually impaired

**Comment by brianteeman** [comment]
Unless I got !y code wrong that's all done

**Comment by mbabker** [comment]
Unless I misread something the tooltip (title attribute) covers things fine for people using screenreaders or other assistive software.  The only thing that's not good is for people not using those and seeing an icon, but repeating the word "Edit" 20 times doesn't seem any better.

**Comment by C-Lodder** [comment]
Tooltips has absolutely nothing to do with this PR. I haven't added any, nor have I removed any. Part 2 of removing them will come in a different PR.

This PR simply adds an icon to the left hand side of the article title and remove the "edit" button from the toolbar, nothing more, nothing less

**Comment by C-Lodder** [comment]
Can someone remove RTC please, as I need to do the rest of the components and this `title` thing needs to be decided on

**Comment by brianteeman** [comment]
For reference this is why you must have a title link for accessibility
https://www.w3.org/TR/WCAG20-TECHS/H33.html

And why links must have unique titles
http://webaim.org/techniques/hypertext/#screen_readers
 Links should make sense out of context. Phrases such as "click here," "more," "click for details," and so on are ambiguous when read out of context

**Comment by dgrammatiko** [comment]
@brianteeman point is about `class="hasTooltip"`, not the `title`

**Comment by C-Lodder** [comment]
@dgt41 `class="hasTooltip"` isn't a problem. We're already in the process of removing tooltips.

The PR adds an edit icon to list item titles and removes the edit icon from the toolbar, nothing else. The tooltip that you see is was already there

**Comment by C-Lodder** [comment]
I'm going to carry on with this PR to remove the edit button from the remaining components. If you guys want to discuss the title any more, you can submit an issue and new PR after

**Comment by C-Lodder** [comment]
Ok, believe they're all done now

**Comment by C-Lodder** [comment]
@wilsonge @brianteeman all happy with this?

**Comment by brianteeman** [comment]
yes

**Pull request merged**